draft-trammell-taps-post-sockets.md

```diff
@@ -818,12 +829,133 @@ func sayHelloWithAssociation(association Association) {
 }
 ~~~~~~~~
 
+## API Dynamics {#sec-dynamics}
+
+As Carriers provide the central entry point to Post, they are key to API
+dynamics. The lifecycle of a carrier is shown in {{fig-carrier-lc}}. Carriers
+are created by active openers by calling Initiate() given a Local and a Remote,
+and by passive openers by calling Listen() given a Local; the .Accept() method
+on the listener Carrier can then be used to create active carriers. By default,
+the underlying Association is automatically created and managed by the
+underlying API. This underlying Association can be accessed by the Carrier's
+.Association() method. Alternately, an association can be explicitly created
+using NewAssociation(), and a Carrier on the association may be accessed or
+initiated by calling the association's .Initiate() method.
+
+Once a Carrier has been created (via Initiate(), Association.Initiate(),
+NewSource(), NewSink(), or Listen()/Accept()), it may be used to send and
+receive Messages. The existence of a Carrier does not imply the existence of
+an active Transient or associated transport-layer connection; these may be
+created when the carrier is, or may be deferred, depending on the network
+environment, configuration, and protocol stacks available.
+
+~~~~~~
+ Listen(local)   Initiate(local,remote)   NewSource(local,remote)
+      |                     |             or NewSink(local)
+  [ Carrier  ]              |                    |
+  [(listener)]              +--------------------+
+      |                     V                    
+  .Accept()-----------> [Carrier] -+----------> .Close()
+                         |     ^   |  close    [ Carrier  ]
+                         |     |   +- event -> [ (closed) ]
+                         |     |          
+              .Association()  .Carriers()
+                         |    .Initiate()
+                         V     |
+                     [Association]   
+                           ^
+                           |
+              NewAssociation(local,remote)
+~~~~~~~
+{: #fig-carrier-lc title="Carrier and Association Life Cycle"}
+
+Access to more detailed information is possible through accessors on Carriers
+and Associations, as shown in {{fig-accessors}}. The set of currently active
+Transients can be accessed through the Carrier's .Transients() methods. The
+active path(s) used by a Transient can be accessed through the Transient's
+.Paths() method, and the set of all paths for which properties are cached by
```

Comment:
Section 2.7 says "a Transient contains ephemeral state for a single transport protocol over a single Path", so either that needs relaxing to a set of Paths, or this needs to be a .Path() method that returns a single Path,

Reply:
the language in 2.7 is old, and doesn't reflect the current thinking that there might be multiple paths behind a PSI (as when using MPTCP unmodified), so I'll change it there.

Reply:
ack